src/Appwrite/Platform/Modules/Functions/Http/Executions/Create.php

```
@@ -180,7 +180,8 @@ public function action(
 
         $version = $function->getAttribute('version', 'v2');
         $runtimes = Config::getParam($version === 'v2' ? 'runtimes-v2' : 'runtimes', []);
-        $spec = Config::getParam('specifications')[$function->getAttribute('specification', APP_COMPUTE_SPECIFICATION_DEFAULT)];
+
+        $spec = Config::getParam('specifications')[$resource->getAttribute('runtimeSpecification', APP_COMPUTE_SPECIFICATION_DEFAULT)];
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$resource'. (undefined)

(UnusedLocalVariable)